### PR TITLE
Update README to be in sync with Package Storage V2

### PR DESCRIPTION
**Pull request by mrodm**
This PR updates the README to update the references to snapshot/staging to be deprecated, as well as review the examples shown to test it according with the latest changes. For instance,  the base docker image used in [elastic-package](https://github.com/elastic/elastic-package) is no longer `docker.elastic.co/package-registry/distribution:snapshot`.

To be merged after the switch to Package Storage V2 is done

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ There are several options to run this for development purposes.
 ### Go command
 
 To use the correct golang version, run:
+- We recommend using [GVM](https://github.com/andrewkroh/gvm), same as done in the CI.
+  This tool allows you to install multiple versions of Go, setting the Go environment in consequence: `eval "$(gvm 1.15.9)"`
```
Comment (jsoriano) [inline README.md:68]: It looks weird to have this recommendation as a list element here.
```suggestion
We recommend using [GVM](https://github.com/andrewkroh/gvm), same as done in the CI.
This tool allows you to install multiple versions of Go, setting the Go environment in consequence: `eval "$(gvm 1.15.9)"`

To use the correct golang version, run:
```

File: README.md [hunk 2]
```diff
@@ -103,7 +103,7 @@ kubectl get deployment -n package-registry
 This will output the list of available deployments. To do a rolling restart of the staging deployment run:
 
 ```
-kubectl rollout restart deployment package-registry-staging-vanilla -n package-registry
+kubectl rollout restart deployment package-registry-v2-vanilla -n package-registry
```
Comment (jsoriano) [inline README.md:106]: Not sure if we should mention this in the public docs :thinking:

File: README.md [hunk 3]
```diff
@@ -148,42 +148,33 @@ These images contain only the package registry, they don't contain any package.
 
 ### Testing with Kibana
 
-The Docker image of Package Registry is just an empty distribution without any packages. To test it with Kibana using
-[elastic-package](https://github.com/elastic/elastic-package), you need to rebuild the snapshot distribution first:
+The Docker image of Package Registry is just an empty distribution without any packages.
+By default, Package Registry has the proxy mode enabled. Thereby, even if there is no package in the
```
Comment (jsoriano) [inline README.md:152]: I think that proxy mode is not enabled by default. It is only in `elastic-package`, with `EPR_FEATURE_PROXY_MODE=true`.
We could tough mention how to enable it here.

Comment (mrodm) [reply README.md]: I've added a new section for the proxy mode feature in the Readme

File: README.md [hunk 4]
```diff
@@ -251,6 +228,22 @@ For example:
...
+You can enable in Package Registry the proxy mode. This mode allows to take into account all the packages
+from other endpoint as part of the responses.
+
+This mode is enabled with the parameter `-feature-proxy-mode=true` (or `EPR_FEATURE_PROXY_MODE` environment variable).
+And it will use by default as proxy endpoint `https://epr.elastic.co`. This endpoint can be customized using the parameter `-proxy-to`
+(or `EPR_PROXY_TO`).
+For example:
+
+```
+package-registry --feature-proxy-mode=true -proxy-mode=https://epr.elastic.co
```
Comment (jsoriano) [inline README.md:243]:
```suggestion
package-registry --feature-proxy-mode=true -proxy-to=https://epr.elastic.co
```

Comment (mrodm) [reply README.md:243]: Ouch! I was checking the parameter spelling in the paragraph, and I missed it here 
Thanks!

**Review by jsoriano — APPROVED** [review]

**Pull request merged**